### PR TITLE
Am/chore/tfhe ntt

### PR DESCRIPTION
**Pull request by IceTDrinker**

### CODE REVIEW

**Review by soonum — COMMENTED** [review]
Huge import again, thanks for handling this tedious task.

File: .github/workflows/m1_tests.yml [hunk 1]
```diff
@@ -64,13 +64,29 @@ jobs:
           make test_fft_no_std_nightly
           # we don't run the js stuff here as it's causing issues with the M1 config
 
+      - name: Run pcc NTT checks
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply .github/workflows/m1_tests.yml:67]: actually the FFT and NTT tests/checks are very quick and the M1 workflow passed on this PR

File: .github/workflows/benchmark_tfhe_ntt.yml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    name: Execute NTT benchmarks in EC2
+    needs: setup-ec2
+    concurrency:
+      group: ${{ github.workflow }}_${{ github.ref }}
+      cancel-in-progress: true
+    runs-on: ${{ needs.setup-ec2.outputs.runner-name }}
+    steps:
+      - name: Get benchmark date
+        run: |
+          echo "BENCH_DATE=$(date --iso-8601=seconds)" >> "${GITHUB_ENV}"
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply .github/workflows/benchmark_tfhe_ntt.yml:51]: I thought It was needed here, I'll check

File: .github/workflows/benchmark_tfhe_ntt.yml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+          COMMIT_DATE="$(git --no-pager show -s --format=%cd --date=iso8601-strict ${{ github.sha }})"
+          COMMIT_HASH="$(git describe --tags --dirty)"
+          python3 ./ci/ntt_benchmark_parser.py target/criterion ${{ env.RESULTS_FILENAME }} \
+          --database concrete_ntt \
+          --hardware "hpc7a.96xlarge" \
+          --project-version "${COMMIT_HASH}" \
+          --branch ${{ github.ref_name }} \
+          --commit-date "${COMMIT_DATE}" \
+          --bench-date "${{ env.BENCH_DATE }}" \
+          --name-suffix avx512
```
[the remark this answers is not in the document]
Comment (soonum) [reply .github/workflows/benchmark_tfhe_ntt.yml:79]: Copy paste error, I wanted to reuse the pattern used for tfhe-fft benchmarks.

File: .github/workflows/benchmark_tfhe_ntt.yml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      cancel-in-progress: true
+    runs-on: ${{ needs.setup-ec2.outputs.runner-name }}
+    steps:
+      - name: Get benchmark details
+        run: |
+          {
+            echo "BENCH_DATE=$(date --iso-8601=seconds)";
+            echo "COMMIT_DATE=$(git --no-pager show -s --format=%cd --date=iso8601-strict ${{ github.sha }})";
+            echo "COMMIT_HASH=$(git describe --tags --dirty)";
+          } >> "${GITHUB_ENV}"
```
Comment (soonum) [inline .github/workflows/benchmark_tfhe_ntt.yml:55]: You need to do that after the checkout, otherwise you'll have an error with `git` commands.

Comment (IceTDrinker) [reply .github/workflows/benchmark_tfhe_ntt.yml:60]: indeed done

**Review by soonum — APPROVED** [review]
Welcome to `tfhe-ntt` crate :rocket:

**Pull request merged**